File: leetcode/L804.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        morseMap = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        bag = set()
        for word in words:
            mlist = [morseMap[ord(c)-97] for c in word]
            mword = ''.join(mlist)
            bag.add(mword)
        return len(bag)

s = Solution()
words = ["gin", "zen", "gig", "msg"];
print(s.uniqueMorseRepresentations(words))

File: leetcode/L804.java
class Solution { 
   public int uniqueMorseRepresentations(String[] words) {
        String[] MORSE = new String[]{
                         ".-","-...","-.-.","-..",".","..-.","--.",
                         "....","..",".---","-.-",".-..","--","-.",
                         "---",".--.","--.-",".-.","...","-","..-",
                         "...-",".--","-..-","-.--","--.."};
        Set<String> bag = new HashSet();
        for (String word: words) {
            StringBuilder mlist = new StringBuilder();
            for(char c: word.toCharArray())
                mlist.append(MORSE[c - 'a']);
            bag.add(mlist.toString());
        }
        return bag.size();
   }
}
